Added section 4.3.7 to DOB, App Service

densellp opened the pull request:
- Added App Service Section to DOB (4.3.7)
- Added Deliverables to be discussed
- Added image to show the response from spring petclinic when it works properly

nsshaddox commented on docs/4-cloud-computing/4.3.7-app-service.md:
```diff
@@ -0,0 +1,97 @@
+
+### Exercise 1: Deploy a Java Application
+
+We will be using liatrio's version of the spring-petclinic for this exercise, its a simple java application that deploys a web app for a virtual pet clinic.
```
Is liatrio supposed to be lowercase?

nsshaddox replied:
Also, replace its with it's.

densellp replied:
resolved

nsshaddox commented on docs/4-cloud-computing/4.3.7-app-service.md:
````diff
@@ -0,0 +1,97 @@
+git clone https://github.com/YOUR_NAME/spring-petclinic.git
+```
+
+!> There is an official repo for the spring-petclinic, we will not be using that since it contains different build test that may not be useful for our purposes and might break the application on build.
````
There is a grammar error here.

-  "we will not be using that since it contains different build test**s**"
-  "we will not be using that since it contains **a** different build test"

nsshaddox commented on docs/4-cloud-computing/4.3.7-app-service.md:
```diff
@@ -0,0 +1,97 @@
+
+### Exercise 2: Deploy a Docker Application
+
+We have seen that the Azure App Service can be useful for deploying a Java application using Azures built in maven plugin, but what if our application doesn't use java? What if it uses something that isn't natively supported by Azure? That's where docker containers come in, they are super versatile and can run a wide range of applications that are otherwise not supported by the Azure App Service  ...
```
"as a majority **of** Engineers"

densellp replied:
it already says that, is there something else that you were trying to point out?

densellp replied:
resolved

nsshaddox commented on docs/4-cloud-computing/4.3.7-app-service.md:
```diff
@@ -0,0 +1,97 @@
+
+#### Scale Application
+
+Application Scaling in essential in balancing the load and giving the application more resources to execute properly. Azure App Service has this feature built in and we can use the Azure CLI to scale up and scale out our service.
```
"Application Scaling **is** essential"

densellp replied:
resolved

densellp commented:
Made some changes to the App Service section after going through again:
* Removed Azure Monitors: they don't really do anything to the current application and are a little more challenging than expected
* Changed deliverables
* Added a note to enable logs otherwise they are not tracked

jburns24 commented on docs/4-cloud-computing/4.3.7-app-service.md:
```diff
@@ -0,0 +1,92 @@
+
+#### Deploy the Java App to App Service
+
+1. Read the above documentation about Azure App Service and use Microsofts official [Maven Plugin](https://learn.microsoft.com/en-us/azure/app-service/quickstart-java?tabs=javase&pivots=platform-linux-development-environment-maven) to deploy a java application.
+2. Make sure the application you deploy can be reached by copying the domain it gives you at the end and pasting that into your web browser.
```
I just gotta say this is such a nice user experience! hats off to M$

jburns24 commented on docs/4-cloud-computing/4.3.7-app-service.md:
```diff
@@ -0,0 +1,92 @@
+
+### Exercise 2: Deploy a Docker Application
+
+We have seen that the Azure App Service can be useful for deploying a Java application using Azures built in maven plugin, but what if our application doesn't use java? What if it uses something that isn't natively supported by Azure? That's where docker containers come in, they are super versatile and can run a wide range of applications that are otherwise not supported by the Azure App Service  ...
```
```suggestion
We have seen that the Azure App Service can be useful for deploying a Java application using Azures built in maven plugin, but what if our application doesn't use Java? What if it uses something that isn't natively supported by Azure? That's where docker containers come in, they are super versatile and can run a wide range of applications that are otherwise not supported by the Azure App Service natively. In this exercise, we will be taking the spring-petclinic and publish a Docker image to be used with the Azure App Service. This demonstrates a more practical use for App Service with the rise of containerization.
```

jburns24 reviewed (approved):
Great new addition to the bootcamp! Thank you both
